src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -376,7 +377,12 @@ class GoogleDataprocDAO(protected val leoServiceAccountEmail: WorkbenchEmail,
 
   /* set the notebook service account as the staging bucket owner */
   override def setStagingBucketOwnership(cluster: LeoCluster): Future[Unit] = {
-    cluster.serviceAccountInfo.notebookServiceAccount match {
+    setBucketOwnership(cluster.serviceAccountInfo.clusterServiceAccount, cluster.googleProject, cluster.clusterName)
+    setBucketOwnership(cluster.serviceAccountInfo.notebookServiceAccount, cluster.googleProject, cluster.clusterName)
```

Comment:
you are dropping the first future on the floor here. you want to use something like future.traverse. also what happens if the cluster SA and notebook SA are the same? you probably want to put them into a set

tldr, something like:

```
Future.traverse ( Set( clusterServiceAccount, notebookServiceAccount ) ) { sa =>
  setTheOwnership(account, project, clusterName)
}
```

Reply:
+1, that'd work. Another nitpick: we could handle the Options here and make `setBucketOwnership` take a `WorkbenchEmail` instead of an Option, since the None case is a no-op. Something like:
```
Future.traverse(Set(clusterServiceAccount, notebookServiceAccount).flatten) { sa =>
  setBucketOwnership(sa, project, clusterName)
}.void
```

Reply:
my inner monologue:

"wait, what does set.flatten do?"
"ohhhhh"
"goddamn."

i like this approach but it deserves a comment along the lines of "flatten any nones out of the set" for the slow-witted (i.e. me)